Kickoff api

holywyvern opened the pull request:
# Summary

- Installed Rails
- Made kickoff branch
- Generated base project with a Postgresql database
- Added test folder to .gitignore

# Trello Card

https://trello.com/c/QqO7DiBm/37-tutorial-git-github

# Notes
I needed a JS environment to work with, because nobody told me wich, I decided to install Node's latest LTS version with NVM
I also needed to install some dependencies for the pg native gem.

# Screenshot

![screenshot from 2018-06-26 19-22-30](https://user-images.githubusercontent.com/8978215/41972168-66592570-79e7-11e8-8949-e5dc3c3eca5a.png)

epintos commented on Gemfile:
```diff
@@ -0,0 +1,50 @@
+source 'https://rubygems.org'
```
Some comments here. In future projects try to leave this file really pretty, adding comments above each gem explaining why you are introducing it. No need to refactor this one that is auto generated, but it would be nice to do that when you add new ones

epintos commented on README.rdoc:
```diff
@@ -0,0 +1,28 @@
+== README
```
In rails there are lots of commands that adds lots of unused files for you. To avoid having all those files in that repo you can make your own code review as soon as you create your pull request. Its a good method to avoid your code reviewer to use lots of time with those little things that you can fix yourself :)

epintos commented on app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,25 @@
+
+  private
+    def find_article
+      @article = Article.find(params[:article_id])
```
Another way of doing this is using memorization

```ruby
def article
    @article ||= Articlle.find(params[:article_id]
end
```

So you just call article, and if you call it again it if will be in the memory variable

epintos commented on app/helpers/application_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module ApplicationHelper
```
Another autogenerated file

epintos commented on app/views/articles/_form.html.slim:
```diff
@@ -0,0 +1,22 @@
+= form_for(@article) do |f|
```
A good practice is to avoid using variables with `@` in partials. Think that this partials can be used in many many views and you have to guess that you need to set the `@articles` variable. So you should set the variable when you call the partial instead

epintos commented on test/test_helper.rb:
```diff
@@ -0,0 +1,10 @@
+ENV['RAILS_ENV'] ||= 'test'
```
We will use rspec for tests. So it would be nice to remove this folder and add it to the gitignore

epintos commented:
@rominariedel Another comment is that the branch is called development and it should be something like `blog` for example

epintos commented:
:+1:

Commit:
# First branch testing

- Made some changes to test branches
- Testing pull requests
- Checking if markdown works on my comment.

Commit:
Created basic APP
# Summary

- Installed gems
- Installed node as a JS environment
- Used rbenv & ruby-build

# Trello card

https://trello.com/c/0D2ayJi2/16-crear-aplicaci%C3%B3n-rails-wbooksapi

Commit:
Added Rubocop at the project

# Summary
- All files inside app folder now have the frozen string magic comment.

# Trello Card

https://trello.com/c/D8K0qyQh/8-configurar-rubocop

Commit:
Revert "Added Rubocop at the project"

This reverts commit 55ea739d993822f47231333aa1f29c64be316b8e.

leamotta commented on README.md:
```diff
@@ -1 +1,7 @@
+
+## First test
+
+Testing a not really messy branch.
```
You can remove this part of the readme

leamotta commented on training-app/config/application.rb:
```diff
@@ -0,0 +1,19 @@
+# you've limited to :test, :development, or :production.
+Bundler.require(*Rails.groups)
+
+module TrainingApp
```
The app should be called WBooks, as stated in the trello card

leamotta reviewed (approved):
There's no need for a folder to contain all the files, every file should start at root level. Since every PR after this will conflict heavily if we change it now, I'm going to approve this, but later we should move everything to the root folder

Commit:
Removed unused info on readme and changed app name
